python/ray/serve/_private/http_util.py

```diff
@@ -150,7 +150,8 @@ class MessageQueue(Send):
     """
 
     def __init__(self):
-        self._message_queue = asyncio.Queue()
+        # NOTE(edoakes): `deque` appends and pops are thread safe in CPython.
```

Comment:
Are there multiple threads using a single `MessageQueue`?

Reply:
There are not. Sorry this was left over from a previous iteration of prototyping. Thread safety is handled using `run_coroutine_threadsafe` so only one asyncio loop/thread is accessing it (see `replica.py`).